src/App.js

```diff
@@ -1,85 +1,151 @@
+    }));
+
+    this.filterTodoList();
+  };
+
+  filterTodoList = () => {
+    this.setState((prevState) => {
+      const { filterIdentifier, originalTodoList } = prevState;
+
+      switch (filterIdentifier) {
+        case 'active':
+          return ({
+            todoList: originalTodoList.filter(todo => !todo.isCompleted),
+          });
+        case 'completed':
+          return ({
+            todoList: originalTodoList.filter(todo => todo.isCompleted),
+          });
+        default:
+          return ({
+            todoList: [...originalTodoList],
+          });
+      }
+    });
+  };
+
+  toggleAllTodosCompleted = () => {
+    this.setState(prevState => ({
+      originalTodoList: prevState.originalTodoList.map(todo => ({
+        ...todo,
+        isCompleted: prevState.originalTodoList.some(t => !t.isCompleted),
+      })),
+    }));
+
+    this.filterTodoList();
+  };
+
+  toggleTodoCompleted = (id) => {
+    this.setState(prevState => ({
+      originalTodoList: prevState.originalTodoList.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            isCompleted: !todo.isCompleted,
+          };
+        }
+
+        return todo;
+      }),
+    }));
+
+    this.filterTodoList();
+  };
 
-      <footer className="footer">
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button type="button" className="clear-completed">
-          Clear completed
-        </button>
-      </footer>
-    </section>
-  );
+  deleteTodo = (id) => {
+    this.setState(prevState => ({
+      originalTodoList: prevState.originalTodoList
+        .filter(todoItem => todoItem.id !== id),
+    }));
+
+    this.filterTodoList();
+  };
+
+  toggleFilterIdentifier = (identifier) => {
+    this.setState({
+      filterIdentifier: identifier,
+    });
+
+    this.filterTodoList();
+  };
+
+  removeCompletedTodos = () => {
+    this.setState(prevState => ({
+      originalTodoList: prevState.originalTodoList
+        .filter(todoItem => !todoItem.isCompleted),
+    }));
+
+    this.filterTodoList();
+  };
+
+  render() {
+    const {
+      todoList,
+      originalTodoList,
+      filterIdentifier,
+    } = this.state;
+    const amountOfActiveTodos = originalTodoList
+      .filter(todo => !todo.isCompleted).length;
+
+    return (
+      <section className="todoapp">
+        <TodoHeader addTodo={this.handleAddTodo} />
+
+        <section className="main">
+          <input
+            type="checkbox"
+            id="toggle-all"
+            className="toggle-all"
+            checked={this.state.todoList.every(todo => todo.isCompleted)}
```

Comment:
Ты же сделала дестуктуризацию state
```suggestion
            checked={todoList.every(todo => todo.isCompleted)}
```